pdp_util/agg.py

```diff
@@ -38,99 +40,126 @@ def __init__(self, dsn, sesh=None):
+        """Fire off pydap requests and return an iterable (from :func:`ziperator`)"""
         req = Request(environ)
         form = req.params
-        climo = True if 'download-climatology' in form else False
+        climo = True if "download-climatology" in form else False
 
         filters = validate_vars(environ)
         stns = get_stn_list(self.session, filters)
 
         ext = get_extension(environ)
         if not ext:
-            return HTTPBadRequest("Requested extension not supported")(environ, start_response)
-                
-        status = '200 OK'
-        response_headers = [('Content-type','application/zip'), ('Content-Disposition', 'filename="pcds_data.zip"')]
+            return HTTPBadRequest("Requested extension not supported")(
+                environ, start_response
+            )
+
+        status = "200 OK"
+        response_headers = [
+            ("Content-type", "application/zip"),
+            ("Content-Disposition", 'filename="pcds_data.zip"'),
+        ]
         start_response(status, response_headers)
-        environ['pydap.handlers.pcic.dsn'] = self.dsn
+        environ["pydap.handlers.pcic.dsn"] = self.dsn
 
-        responders = chain(get_all_metadata_index_responders(self.session, stns, climo),
-                           get_pcds_responders(self.dsn, stns, ext, get_clip_dates(environ), environ)
-                           )
+        responders = chain(
+            get_all_metadata_index_responders(self.session, stns, climo),
+            get_pcds_responders(self.dsn, stns, ext, get_clip_dates(environ), environ),
+        )
 
         return ziperator(responders)
 
+
 def ziperator(responders):
-    '''This method creates and returns an iterator which yields bytes for a :py:class:`ZipFile` that contains a set of files from OPeNDAP requests. The method will spool the first one gigabyte in memory using a :py:class:`SpooledTemporaryFile`, after which it will use disk.
+    """This method creates and returns an iterator which yields bytes for a :py:class:`ZipFile` that contains a set of files from OPeNDAP requests. The method will spool the first one gigabyte in memory using a :py:class:`SpooledTemporaryFile`, after which it will use disk.
 
-       :param responders: A list of (``name``, ``generator``) pairs where ``name`` is the filename to use in the zip archive and ``generator`` should yield all bytes for a single file.
-       :rtype: iterator
-    '''
-    with SpooledTemporaryFile(1024*1024*1024) as f:
-        yield 'PK' # Response headers aren't sent until the first chunk of data is sent.  Let's get this repsonse moving!
-        z = ZipFile(f, 'w', ZIP_DEFLATED)
+    :param responders: A list of (``name``, ``generator``) pairs where ``name`` is the filename to use in the zip archive and ``generator`` should yield all bytes for a single file.
+    :rtype: iterator
+    """
+    with SpooledTemporaryFile(1024 * 3) as f:
```

Comment:
Oooops! I didn't catch this before you merged, but this change is not correct. I think you meant `1024 ** 3`. Mind fixing that quickly?

Reply:
Yes thanks!